File: 2023/6.rb
input = File.read(File.basename(__FILE__).gsub('rb', 'input'))
input = input.split("\n")

time = input[0].split[1..].map(&:to_i)
distance = input[1].split[1..].map(&:to_i)

total = 1
(0...time.size).each do |i|
  ways = 0
  (0...time[i]).each do |t|
    ways += 1 if distance[i] < (time[i] - t) * t
  end
  total *= ways
end

p total

time = time.join.to_i
distance = distance.join.to_i
ways = 0
(0...time).each do |t|
  ways += 1 if distance < (time - t) * t
end

p ways

File: 2023/6.py
input = open(f"{__file__.replace('py', 'input')}").read().strip().split("\n")

time, distance = input
time = [int(x) for x in time.split()[1::]]
distance = [int(x) for x in distance.split()[1::]]

total = 1
for i, _ in enumerate(time):
    ways = 0
    for t in range(0, time[i]):
        if distance[i] < (time[i] - t) * t:
            ways += 1
    total *= ways
print(total)

time = int("".join([str(x) for x in time]))
distance = int("".join([str(x) for x in distance]))
ways = 0
for t in range(0, time):
    if distance < (time - t) * t:
        ways += 1
print(ways)
